### DISCUSSION TITLE
Using server middleware to redirect www to non www no longer works as it did in Nuxt 2 and I cannot find documentation to advise otherwise

### DISCUSSION THREAD

**Discussion by toddpadwick** [nuxt/nuxt#18435] [Questions]
### Environment

------------------------------
- Operating System: `Darwin`
- Node Version:     `v14.19.0`
- Nuxt Version:     `3.0.0`
- Nitro Version:    `1.0.0`
- Package Manager:  `npm@6.14.16`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

### Reproduction

If this is just a documentation issue then this isn't required.

### Describe the bug

With Nuxt 2, I used server middleware to redirect www to non www as follows which worked fine:
```
module.exports = function (req, res, next) {
  const host = req.headers.host
  const url = req.url
  const env = process.env.ENVIRONMENT
  const mainDomain = 'www.mydomain.com'
  const forceDomain = 'https://'+mainDomain

  if (env === 'production' && host !== mainDomain) {
    res.writeHead(301, { Location: forceDomain + url })
    return res.end()
  }

  return next()
}

```

I tried doing the same with Nuxt 3, by saving this file in /server/middleware as per docs but I get `host is not defined` or `next is not defined`. 
I can't see any information on this and whether this is either a bug, or something that has changed in the docs.  If this is a documentation issue please could you advise on the new best practice method for handling this?

**Comment by cpreston321** [comment]
@toddpadwick 

Here is a working prototype of what you are trying to achieve. Let me know if you need any help. 

https://stackblitz.com/edit/github-hncs2a?file=server/middleware/redirect.ts

Also if you want to check out what it's doing behind the scences please check out [Nitro Docs](https://nitro.unjs.io/) & H3 Repo [here](https://github.com/unjs/h3).

Thanks 💚

**Comment by jmcclung3509** [comment]
@toddpadwick   Hi.  Did you figure this out?  I am having the same problem and so far nothing I have tried has worked.
